src/plugins/ui_core_views/cell_evaluation/evaluator.ts

```diff
@@ -137,6 +138,10 @@ export class Evaluator {
     );
     this.compilationParams.evalContext.updateDependencies = this.updateDependencies.bind(this);
     this.compilationParams.evalContext.addDependencies = this.addDependencies.bind(this);
+    this.compilationParams.evalContext.lookupCaches = {
+      forwardSearch: new Map<any, Map<CellValue, number>>(),
+      reverseSearch: new Map<any, Map<CellValue, number>>(),
+    };
```

Comment:
```suggestion
    this.compilationParams.evalContext.caches= {};
```
nitpick. And the the function can just set `this.cache = new Map()` That way each function can setup its cache as needed, and the evaluator doesn't have to know which function needs a cache.

Reply:
I am looking for an even more general solution which will automatically include all calculation times that are too long at the scale of the evaluation. We can also discuss it again, I would be curious to have your opinion :)